Sources/TuistSupport/Utils/FileHandler.swift

```diff
@@ -66,7 +66,12 @@ public protocol FileHandling: AnyObject {
     func write(_ content: String, path: Path.AbsolutePath, atomically: Bool) throws
     func locateDirectoryTraversingParents(from: Path.AbsolutePath, path: String) -> Path.AbsolutePath?
     func locateDirectory(_ path: String, traversingFrom from: Path.AbsolutePath) throws -> Path.AbsolutePath?
-    func files(in path: Path.AbsolutePath, nameFilter: Set<String>?, extensionFilter: Set<String>?) -> Set<Path.AbsolutePath>
+    func files(
+        in path: Path.AbsolutePath,
+        filter: ((URL) -> Bool)?,
+        nameFilter: Set<String>?,
+        extensionFilter: Set<String>?
+    ) -> Set<Path.AbsolutePath>
```

Comment:
Let's create another method without `filter` and then add a default parameter in the `FileHandler` implementation

Reply:
A default parameter doesn't work for protocol conformance of a func without the filter argument. The protocol has now two files(in) functions one with a filter and one without. I extended the protocol to add a default implementation where I just pass nil to the filter. I think that should give the same result.